base/daemon/SentryWriter.cpp

```diff
@@ -0,0 +1,244 @@
+            case SENTRY_LEVEL_DEBUG:
+                logger->debug(buffer);
+                break;
+            case SENTRY_LEVEL_INFO:
+                logger->information(buffer);
+                break;
+            case SENTRY_LEVEL_WARNING:
+                logger->warning(buffer);
+                break;
+            case SENTRY_LEVEL_ERROR:
+                logger->error(buffer);
+                break;
+            case SENTRY_LEVEL_FATAL:
+                logger->fatal(buffer);
+                break;
+        }
+    }
+}
+}
+#endif
+
+void SentryWriter::initialize(Poco::Util::LayeredConfiguration & config)
+{
+#if USE_SENTRY
+    bool enabled = false;
+    bool debug = config.getBool("send_crash_reports.debug", false);
+    auto * logger = &Poco::Logger::get("SentryWriter");
+    if (config.getBool("send_crash_reports.enabled", false))
+    {
+        if (debug || (strlen(VERSION_OFFICIAL) > 0))
+        {
+            enabled = true;
+        }
+    }
+    if (enabled)
+    {
+        const std::string & endpoint
+            = config.getString("send_crash_reports.endpoint", "https://6f33034cfe684dd7a3ab9875e57b1c8d@o388870.ingest.sentry.io/5226277");
+        const std::string & temp_folder_path
+            = config.getString("send_crash_reports.tmp_path", config.getString("tmp_path", Poco::Path::temp()) + "sentry/");
+        Poco::File(temp_folder_path).createDirectories();
+
+        sentry_options_t * options = sentry_options_new();
+        sentry_options_set_release(options, VERSION_STRING);
+        sentry_options_set_logger(options, &sentry_logger);
+        if (debug)
+        {
+            sentry_options_set_debug(options, 1);
+        }
+        sentry_options_set_dsn(options, endpoint.c_str());
+        sentry_options_set_database_path(options, temp_folder_path.c_str());
+        if (strstr(VERSION_DESCRIBE, "-stable") || strstr(VERSION_DESCRIBE, "-lts"))
+        {
+            sentry_options_set_environment(options, "prod");
+        }
+        else
+        {
+            sentry_options_set_environment(options, "test");
+        }
+
+        const std::string & http_proxy = config.getString("send_crash_reports.http_proxy", "");
+        if (!http_proxy.empty())
+        {
+            sentry_options_set_http_proxy(options, http_proxy.c_str());
+        }
+
+        int init_status = sentry_init(options);
+        if (!init_status)
+        {
+            initialized = true;
+            anonymize = config.getBool("send_crash_reports.anonymize", false);
+            const std::string& anonymize_status = anonymize ? " (anonymized)" : "";
+            LOG_INFO(
+                logger,
+                "Sending crash reports is initialized with {} endpoint and {} temp folder{}",
+                endpoint,
+                temp_folder_path,
+                anonymize_status);
+        }
+        else
+        {
+            LOG_WARNING(logger, "Sending crash reports failed to initialized with {} status", init_status);
```

Comment:
failed to initialized
->
failed to initialize

Reply:
Is there something like strerror?

Reply:
> Is there something like strerror?

@alexey-milovidov I don't think so, it looks like it logs something and exits with 1.